shared/app/nav.native.js

```diff
@@ -13,22 +13,22 @@ import {
   NativeStatusBar,
 } from '../common-adapters/index.native'
 import {NavigationActions, type NavigationAction} from 'react-navigation'
-import {chatTab, loginTab, peopleTab, folderTab, settingsTab, type Tab} from '../constants/tabs'
+import {chatTab, loginTab, peopleTab, folderTab, settingsTab} from '../constants/tabs'
 import {compose} from 'recompose'
 import {connect, type TypedState} from '../util/container'
-import {globalColors, globalStyles, statusBarHeight} from '../styles/index.native'
+import {globalColors, globalStyles, statusBarHeight, styleSheetCreate} from '../styles/index.native'
 import {addSizeListener} from '../styles/status-bar'
 import * as I from 'immutable'
 import {isIOS, isIPhoneX} from '../constants/platform'
-import {navigateTo, navigateUp, switchTo} from '../actions/route-tree'
+import {navigateUp} from '../actions/route-tree'
 import {tabBarHeight} from './tab-bar/index.render.native'
 import {type Props, type OwnProps} from './nav'
 import {type RouteRenderStack, type RenderRouteResult} from '../route-tree/render-route'
 import {makeLeafTags} from '../route-tree'
 
 type CardStackShimProps = {
   mode?: 'modal',
-  renderRoute: (route: RenderRouteResult, isActiveRoute: boolean, shouldRender: boolean) => any,
```

Comment:
the activeRoute concept was there to reduce work on components that aren't currently visible.
really we should ensure our mapStateToProps calls are cheap or unmount these expensive components